src/detectors/typeChecker/FileLinter.ts

```diff
@@ -211,6 +212,89 @@ export default class FileLinter {
 		});
 	}
 
+	/**
+	 * Extracts & builds object literal
+	 *
+	 * @param node
+	 * @returns {object}
+	 */
+	extractPropsRecursive = (node: ts.ObjectLiteralExpression) => {
+		type propsType = Record<string, {value: null | boolean | string | propsType; node: ts.Node}>;
+		const properties = Object.create(null) as propsType;
+
+		node.properties?.forEach((prop) => {
+			if (!ts.isPropertyAssignment(prop) || !prop.name) {
+				return;
+			}
+
+			const key = prop.name.getText();
+			if (prop.initializer.kind === ts.SyntaxKind.FalseKeyword) {
+				properties[key] = {value: false, node: prop.initializer};
+			} else if (prop.initializer.kind === ts.SyntaxKind.NullKeyword) {
+				properties[key] = {value: null, node: prop.initializer};
+			} else if (ts.isObjectLiteralExpression(prop.initializer) && prop.initializer.properties) {
+				properties[key] = {value: this.extractPropsRecursive(prop.initializer), node: prop.initializer};
+			} else if ((ts.isIdentifier(prop.initializer) ||
+			ts.isNumericLiteral(prop.initializer) ||
+			ts.isStringLiteral(prop.initializer)) &&
+			prop.initializer.text) {
+				properties[key] = {value: prop.initializer.getText(), node: prop.initializer};
+			}
+		});
+		return properties;
+	};
+
+	getSymbolModuleDeclaration(symbol: ts.Symbol) {
+		let parent = symbol.valueDeclaration?.parent;
+		while (parent && !ts.isModuleDeclaration(parent)) {
+			parent = parent.parent;
+		}
+		return parent;
+	}
+
+	analyzeLibInitCall(node: ts.CallExpression) {
+		const nodeExp = (ts.isPropertyAccessExpression(node.expression) ||
```

Comment:
Do we want to also detect and analyze cases like the following? (CC: @matz3)

```js
const myInit = Library.init;
myInit({
    apiVersion: 1
});
```

If yes, this part needs to be adapted to allow for call expressions that are not based on a Property- or ElementAccessExpression.

In any case, I think we should add a test for such a scenario (either positive or negative).

Reply:
Then we (potenitally) might have also destructors and all the implications that come with them: 
``js
const {init} = Library;
init({
    apiVersion: 1
});
// OR
const {init: myInit} = Library;
myInit({
    apiVersion: 1
});
``

Reply:
For now, we should only detect the simple case via `Lib.init` (current implementation).

Reply:
I already tried the above scenarios and most of them seem to be feasible and easy to solve with couple lines of code. 
Only destructuring in `sap.ui.define` would require more efforts here.

Reply:
At this point, `sap.ui.define` should already be transpiled to ESM import/export declarations. That's why you can ask the tsc checker for the type of `Library` and it resolves it via the import.